services/packagist/packagist-downloads.service.js

```diff
@@ -1,123 +1,86 @@
 'use strict'
 
-const LegacyService = require('../legacy-service')
-const { makeBadgeData: getBadgeData } = require('../../lib/badge-data')
 const { metric } = require('../../lib/text-formatters')
-const {
-  downloadCount: downloadCountColor,
-} = require('../../lib/color-formatters')
+const { downloadCount } = require('../../lib/color-formatters')
+const { keywords, BasePackagistService } = require('./packagist-base')
 
-const keywords = ['PHP']
+module.exports = class PackagistDownloads extends BasePackagistService {
+  static get route() {
+    return {
+      base: 'packagist',
+      pattern: ':type(dm|dd|dt)/:user/:repo',
+    }
+  }
 
-// This legacy service should be rewritten to use e.g. BaseJsonService.
-//
-// Tips for rewriting:
-// https://github.com/badges/shields/blob/master/doc/rewriting-services.md
-//
-// Do not base new services on this code.
-module.exports = class PackagistDownloads extends LegacyService {
-  static get category() {
-    return 'downloads'
+  static get defaultBadgeData() {
+    return {
+      label: 'downloads',
+    }
   }
 
-  static get route() {
+  async handle({ type, user, repo }) {
+    const {
+      package: { downloads },
+    } = await this.fetch({ user, repo })
+
+    switch (type) {
+      case 'dm':
+        return this.constructor.render({
+          downloads: downloads.monthly,
+          ending: '/month',
```

Comment:
Instead of computing `ending` here, could you pass the interval to `render` and handle the suffix there?